src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -613,44 +630,107 @@ namespace
         const bool              use_max_proc_maps,
         const TimeValue         time,
         ObjectMap&              object_map,
-        MaterialMap&            material_map)
+        MaterialMap&            material_map,
+        AssemblyMap&            assembly_map)
     {
         // Retrieve the geometrical object referenced by this node.
         Object* object = node->GetObjectRef();
 
-        // Check if we already generated the corresponding appleseed object.
-        const ObjectMap::const_iterator it = object_map.find(object);
+        // Compute the transform of this instance.
+        const asf::Transformd transform =
+            asf::Transformd::from_local_to_parent(
+                to_matrix4d(node->GetObjTMAfterWSM(time)));
 
-        if (it == object_map.end())
+        if (get_generate_assembly(object, time))
         {
-            // The appleseed objects do not exist yet, create and instantiate them.
-            const auto object_infos = create_mesh_objects(assembly, node, time);
-            object_map.insert(std::make_pair(object, object_infos));
-            for (const auto& object_info : object_infos)
+            std::string object_assembly_name = wide_to_utf8(node->GetName());
```

Comment:
Make const.

Reply:
Maybe just `assembly_name`?

Reply:
Can rename but can't make const, it assigned something after that line and further below in else clause